dynamic_programming/range_maximum_query.py

```diff
@@ -0,0 +1,100 @@
+"""
+Range Maximum Query Problem:
+
+    Given an array of integers and q queries,
+    For each query find the maximum value in the range [l, r).
+
+    The array is 0-indexed and the queries are 0-indexed.
+
+    More details:
+        https://www.geeksforgeeks.org/range-maximum-query-using-sparse-table/
+"""
+
+
+class RangeMaximumQuery:
+    def __init__(self, array: list) -> None:
+        """
+        Initialize RangeMaximumQuery with given array.
+
+        Parameters:
+            array: list[int]
+
+        Example:
+        >>> rmq = RangeMaximumQuery([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
+        """
+        self.array = array
+        self.size = len(array)
```

Comment:
Please define a [`__len__()` function](https://docs.python.org/3/reference/datamodel.html#object.__len__) instead of using a `size` variable.